tests/cpp/test_sharding.cpp

```diff
@@ -62,6 +64,41 @@ TEST_F(ShardingTest, PropagateSharding) {
   EXPECT_TRUE(getTvsWithDifferentSharding(a, tvs).empty());
 }
 
+TEST_F(ShardingTest, ShardedAllocationDomain) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* a = makeContigTensor(3);
+  TensorView* b = makeContigTensor(3);
+  TensorView* c = add(a, b);
+  TensorView* d = sum(c, {1});
+
+  DeviceMesh mesh = DeviceMesh::createForNumDevices(3);
+  for (auto tv : {a, b, c, d}) {
+    tv->setDeviceMesh(mesh);
+  }
+  a->axis(1)->parallelize(ParallelType::DIDx);
+  c->axis(1)->parallelize(ParallelType::DIDx);
+  fusion.addInput(a);
+  fusion.addInput(b);
+  fusion.addOutput(d);
+
+  propagateShardings(&fusion);
+  insertReshardings(&fusion);
+  insertShardedAxisReordering(&fusion);
+  setShardedAllocationDomain(&fusion);
+  for (auto expr : fusion.exprs()) {
```

Comment:
How do you like making the check more specific? For example, is it possible to check the following:
* I'd expect there's only one resharding Expr, which is a sum
* I'd also expect the input of that Expr has DID as the first IterDomain in the containing allocation domain.